base/tuple.jl

```diff
@@ -28,7 +28,21 @@ getindex(t::Tuple, b::AbstractArray{Bool,1}) = length(b) == length(t) ? getindex
 getindex(t::Tuple, c::Colon) = t
 
 # returns new tuple; N.B.: becomes no-op if i is out-of-bounds
+
+"""
+    setindex(c::Tuple, v, i::Integer)
+
+Creates a new tuple similar to `x` with the value at index `i` set to `v`.
+An out-of-bound `i` makes it a no-op.
```

Comment:
is that intentional? Seems like odd behavior. I would assume it should boundscheck by default?

Reply:
No idea, @vtjnash was the last to edit the line

Reply:
Was introduced in #20154 as a simple helper function.  The behavior is very much intentional there, but perhaps we should see if we can add a `@boundscheck` block without sacrificing performance if we're going to move towards this being more documented.

Reply:
Wouldn't it be considered as a breaking change then (replacing noop with BoundsError)?

Reply:
Not really as it was neither exported nor documented.  That's why if we document it, it'd be good to make sure the semantics are what we want.

Reply:
FYI StaticArrays.jl does bound check with `setindex` https://github.com/JuliaArrays/StaticArrays.jl/blob/2583967e282b21fd8f8a4e2fc306efc672b4d83d/src/deque.jl#L73-L86